hooks/read-package-hook/src/createPeerDependencyPatcher.ts

```diff
@@ -44,6 +46,22 @@ export function createPeerDependencyPatcher (
 
       pkg.peerDependencies![peerName] = currentVersions.join(' || ')
     }
+
+    let overrides: ReturnType<typeof parseOverrides>
+    try {
+      overrides = parseOverrides(peerDependencyRules.allowedVersions ?? {})
```

Comment:
you need to move parsing of the overrides out of the loop and the arrow function. There's no need to parse it over and over because it doesn't change.